main/acl.c

```diff
@@ -872,18 +872,29 @@ enum ast_acl_sense ast_apply_ha(const struct ast_ha *ha, const struct ast_sockad
 	return res;
 }
 
-static int resolve_first(struct ast_sockaddr *addr, const char *name, int flag,
-			 int family)
+static int resolve_match_or_first(struct ast_sockaddr *addr, const char *name, int flag,
+								  int family, struct ast_sockaddr *preference)
 {
 	struct ast_sockaddr *addrs;
 	int addrs_cnt;
 
 	addrs_cnt = ast_sockaddr_resolve(&addrs, name, flag, family);
 	if (addrs_cnt > 0) {
-		if (addrs_cnt > 1) {
+		struct ast_sockaddr *resolved = &addrs[0];
+
+		if (preference != NULL) {
```

Comment:
We just use if (preference) { in our coding guidelines

Reply:
Updated